Sources/Fuzzilli/Core/ProgramBuilder.swift

```diff
@@ -377,31 +377,25 @@ public class ProgramBuilder {
 
     // This expands and collects types for arguments in function signatures.
     private func prepareArgumentTypes(forSignature signature: FunctionSignature) -> [Type] {
-        var parameterTypes = signature.inputTypes
+        var callerTypes = signature.callerTypes
```

Comment:
I think this function should be rewritten to use `Parameter` directly, something like this:
```swift
var argumentTypes = [Type]()
for param in signature {
    if param.isOptional {
        // It's an optional argument, so stop here in some cases
        ...
    }
    if param.isRest {
        // "Unroll" the rest parameter
        for _ in 0..<Int.random(in: 0...5) {
            argumentTypes.append(param.callerType)
        }
        // Rest parameter must be the last one
        break
    }
    argumentTypes.append(param.callerType)
}
```